report/2.Analysis/EmbeddedPlatforms.tex

```diff
@@ -0,0 +1,78 @@
+
+As image recognition is an expensive computation, the current idea is to offload this to an external server or computer, which will compute the current object position and send that to the embedded device.
+
+The current idea is to use the Lego Mindstorms platform for the robotics part of the solution, and use a single-board computer for image object tracking and the machine learning model training.
+
+\subsection{Lego Mindstorms}
+Lego Mindstorms is a programmable computer kit, made by Lego.
+The platform is based around a central control computer, called the Intelligent Brick.
+It also includes a variety of sensors, motors and connection cables.
+Each generation further improves on the concept, adding more computational power and better sensors.
+
+The Lego Mindstorm platform, comes with a graphical drag and drop interface.
+However, various different compilers for many different languages exist.
+As such the choice between different generations, does not come down to language preferences.
+
+The choices for the 5th semester are the second generation, the Lego Mindstorms NXT 2.0 kit and the third generation, the Lego Mindstorms EV3 kit.
+
+\subsubsection{Lego Mindstorms NXT 2.0}
+The Lego Mindstorms NXT 2.0 (NXT) is the second generation Lego Mindstorms kit.
+The NXT Intelligent Brick is a micro computer based on the 32-bit ARM7 microprocessor, with 256 kbytes of flash memory and 64 kbytes of RAM\cite{nxt2userguide}.
+It includes a variety of sensors, such as touch, sound, light and ultrasonic (vision) sensors.
+On top of this, the consumer kit also includes servo motors and lamps.
+
+A potential major disadvantage of the NXT 2.0, is that the platform is rather old.
+This means that problems could arise, especially in regards to compatability with newer platforms.
+This will be further elaborated on in a later section
+
+\subsubsection{Lego Mindstorms EV3}
+The  Lego Mindstorms EV3 (EV3) is the third generation of the Lego Mindstorms line.
+The EV3 mainly improves on the specifications of the second generation, adding a significantly more powerful 300MHz ARM9 processor, running Linux, 16 MB of flash memory, which is extendable with a microSDHC card and 64MB of RAM\cite{ev3userguide}.
+On top of this, it also adds remote control and WiFi capabilities, as well as more recently updated compilers.
+
+The primary advantage of the EV3 is that its software is up to date and the power it brings to the table.
+The power is however also its primary disadvantage.
+As the goal of this project is to design an embedded real time system, a too powerful computer, could potentially mask some of the challenges that normally come with the development of an embedded system.
+
+%Don't know how relevant this will actually be.
+\subsection{Single-board computers}
+Besides the Lego Mindstorms platform, which is the recommended platform for this semester, there is also the option of using single-board computers, such as the Raspberry Pi or the Arduino.
```

Comment:
disse type sætninger jeg lidt vil væk fra. "godt for semester" skal ikke være 100% grundlag, men det er et godt argument i en konklusion i kapitlet.